### PR TITLE
[hyperion] Fixes #6377 hyperion NPE

### LINKED ISSUE #6377: [hyperion] Binding not working due to NPE in HyperionNgHandler
## Expected Behavior
Thing shows up as online, however Im unable to control my Hyperion-ng server using OpenHAB.
Tested using both thing and item files as well as PaperUI.

## Current Behavior
Following shows up in the log:
```
2019-11-11 19:18:00.826 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 

java.lang.NullPointerException: null

	at org.openhab.binding.hyperion.internal.handler.HyperionNgHandler.updateHyperion(HyperionNgHandler.java:260) ~[?:?]

	at org.openhab.binding.hyperion.internal.handler.HyperionNgHandler.handleServerInfoResponse(HyperionNgHandler.java:178) ~[?:?]

	at org.openhab.binding.hyperion.internal.handler.HyperionNgHandler$1.run(HyperionNgHandler.java:103) ~[?:?]

	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:?]

	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:?]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:?]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:?]

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]

	at java.lang.Thread.run(Thread.java:748) [?:?]
```

## Steps to Reproduce (for Bugs)
1. Install binding
2. Set up Thing
3. Try to control Hyperion-ng

## Your Environment

* Version used: 2.5.0M4 (Both OpenHAB and Hyperion Binding)
* Environment name and version (openjdk version "1.8.0_232")
OpenJDK Runtime Environment (Zulu 8.42.0.23-CA-linux64) (build 1.8.0_232-b18)
OpenJDK 64-Bit Server VM (Zulu 8.42.0.23-CA-linux64) (build 25.232-b18, mixed mode)
* Operating System and version (Ubuntu 18.04.3 LTS):

### PR DESCRIPTION
**Pull request by posseydon6891**
Incorporate changes to HyperionNG serverInfo API - alpha6.
HyperionEnabled state channel is now determined by special component
state "ALL". See: https://docs.hyperion-project.org/en/json/ServerInfo.html

Recent community discussion:
https://community.openhab.org/t/multiple-state-issues-with-hyperion-ng-binding/97747

Signed-off-by: Łukasz Serwicki <lserwicki@gmail.com>

### CODE REVIEW

**Comment by cpmeister** [comment]
Would this change break setups using an older hyperion server?

**Comment by posseydon6891** [comment]
It shouldn't break old stable setups using hyperion V1 as only NG handler is modified.

As for NG server, there are 6 official pre-releases: https://github.com/hyperion-project/hyperion.ng/releases 

Apart from newest alpha.6, I checked also alpha.1 to test if it is working: 
Screenshot_20200612_122121

The "special" component `ALL` code is there and can be seen in API: [serverInfo-alpha1.txt](https://github.com/openhab/openhab-addons/files/4770194/serverInfo-alpha1.txt)
So releases alpha.1 to alpha.6 should not break. 

In case somebody compiled hyperion NG from source before the first release, it depends how old version was used. I tried to check in hyperion NG repo and code related to COMP_ALL is there since commit `d762aa2f3e370b857f501158261ce2a1ba6a5b3f`

Screenshot_20200612_130243

Maybe Daniel (@tavalin) can have a look if it makes sense.

**Comment by cpmeister** [comment]
I mean, I assume that this binding worked at some point with some older version of the hyperion server.
Or can you confirm that this particular feature has never worked?

**Comment by cpmeister** [comment]
If you can find a way to keep support for the old way while adding support for the new then I'd see no problem with merging this.

**Comment by tavalin** [comment]
I haven't looked through the code and as mentioned don't currently have Hyperion running to test the new changes. 

The only thing I would like to see is that the fix doesn't suddenly break the binding for people running versions of hyperion.ng on the old API.

**Comment by posseydon6891** [comment]
ok fair points, I modified the code. Now binding can still use old API format to set server state, based on response content.

Whatever used to work before should also work after this change.

File: bundles/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/handler/HyperionNgHandler.java [hunk 1]
```diff
@@ -173,10 +173,12 @@ public void dispose() {
     protected void handleServerInfoResponse(NgResponse response) {
         NgInfo info = response.getInfo();
         if (info != null) {
-            // update Hyperion
-            Hyperion hyperion = info.getHyperion();
-            updateHyperion(hyperion);
-
+            // update Hyperion, older API compatibility
+            if(info.getHyperion() != null) {
+                Hyperion hyperion = info.getHyperion();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/handler/HyperionNgHandler.java:178]:
```suggestion
            Hyperion hyperion = info.getHyperion();
            if (hyperion != null) {
                
```

File: bundles/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/handler/HyperionNgHandler.java [hunk 2]
```diff
@@ -295,6 +297,9 @@ private void updateComponents(List<Component> components) {
                 case COMPONENT_LEDDEVICE:
                     updateState(CHANNEL_LEDDEVICE, componentState);
                     break;
+                case COMPONENT_ALL:
```
Comment (cpmeister) [inline bundles/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/handler/HyperionNgHandler.java:300]: How does the ALL component only correspond to the hyperion enabled channel? Is there no separate enabled channel?

Comment (posseydon6891) [reply bundles/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/handler/HyperionNgHandler.java:300]: According to [Hyperion docs](https://docs.hyperion-project.org/en/json/ServerInfo.html), sending on/off to special component "ALL":
- saves/restores current state of every component,
- toggles every component on/off,
effectively disabling/enabling hyperion.

Also I double checked by looking at their standalone app which comes with Hyperion:
```
$ hyperion-remote --help | grep -e Activate -e Deactivate
  --off                                              Deactivates hyperion
  --on                                               Activates hyperion
```

What this app do is actually calling API for component "ALL":
```
$ strace -s 9999 -f -e trace=network,write hyperion-remote --off -a 192.168.0.203:19444 
...
write(2, "Connected to: \"192.168.0.203:19444\"\n", 36Connected to: "192.168.0.203:19444"
...
write(2, "Disable Component \"ALL\"\n", 24Disable Component "ALL"
...
write(5, "{\"command\":\"componentstate\",\"componentstate\":{\"component\":\"ALL\",\"state\":false}}\n", 80) = 80
...
```

Basically, there is no dedicated disable/enable channel except API shorthand for disabling/enabling ALL components.

State updates for HyperionEnabled channel look like this in OH logs:
Screenshot_20200615_195229
